pkg/assembler/backends/neo4j/certifyBad.go

```diff
@@ -117,7 +122,7 @@ func (c *neo4jClient) CertifyBad(ctx context.Context, certifyBadSpec *model.Cert
 		aggregateCertifyBad = append(aggregateCertifyBad, result.([]*model.CertifyBad)...)
 	}
 
-	if certifyBadSpec.Source != nil || queryAll {
+	if queryAll || certifyBadSpec.Subject != nil && certifyBadSpec.Subject.Source != nil {
```

Comment:
nit: i like to have brackets whenever there's a mix of ands and ors since left association is a bit weird to think about - or maybe it's just me

Reply:
sure I can add the brackets